drivers/sensor/sngcja5/sngcja5.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023 Panasonic Industrial Devices Europe GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT panasonic_sngcja5
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/__assert.h>
+
+#include <zephyr/logging/log.h>
+
+#include "sngcja5.h"
+
+LOG_MODULE_REGISTER(SNGCJA5, CONFIG_SENSOR_LOG_LEVEL);
+
+static int read_bytes(const struct i2c_dt_spec *dev, uint8_t addr, uint8_t *data,
+		      uint32_t num_bytes)
+{
+	struct i2c_msg msgs[2];
+
+	msgs[0].buf = &addr;
+	msgs[0].len = 1U;
+	msgs[0].flags = I2C_MSG_WRITE;
+
+	msgs[1].buf = data;
+	msgs[1].len = num_bytes;
+	msgs[1].flags = I2C_MSG_RESTART | I2C_MSG_READ | I2C_MSG_STOP;
+
+	return i2c_transfer_dt(dev, &msgs[0], 2);
+}
```

Comment:
Just use `i2c_burst_read_dt` instead

Reply:
Pushed as requested in latest pull request update.